src/iexamine.cpp

```diff
@@ -2280,6 +2280,172 @@ void iexamine::tree_hickory(player &p, const tripoint &examp)
     none( p, examp );
 }
 
+item_location maple_tree_sap_container() {
+    return g->inv_map_splice( []( const item &it ) { 
+        return (
+            it.is_bucket() ||
+            it.is_watertight_container()
+            ) && (
+            it.is_container_empty() || (
+                !it.is_container_full() && 
+                it.contents[0].type->id == "maple_sap"
+                )
+            );
+        }, _( "Which container:" ), PICKUP_RANGE );
+}
+
+void iexamine::tree_maple(player &p, const tripoint &examp)
+{
+    if( !p.has_items_with_quality( "DRILL", 1, 1 ) ) {
+        add_msg( m_info, _( "You need a tool to drill the crust to tap this maple tree." ) );
+        return;
+    }
+
+    if( !p.has_items_with_quality( "HAMMER", 1, 1 ) ) {
+        add_msg( m_info, _( "You need a tool to hammer the spile into the crust to tap this maple tree." ) );
+        return;
+    }
+
+    const inventory &crafting_inv = p.crafting_inventory();
+
+    if( !crafting_inv.has_amount( "maple_tree_spile", 1 ) ) {
+        std::string spile_name = item( "maple_tree_spile", 0 ).tname( 1 );
+        add_msg( m_info, _( "You need a %s to tap this maple tree." ), spile_name.c_str() );
+        return;
+    }
+
+    std::vector<item_comp> comps;
+    comps.push_back( item_comp( "maple_tree_spile", 1 ) );
+    p.consume_items( comps );
+
+    p.moves -= 200;
```

Comment:
`p.mod_moves( -200 )` is slightly better here since we want a good encapsulation.